src/Marten/Services/UnitOfWork.cs

```diff
@@ -342,7 +336,73 @@ public void Eject<T>(T document)
             {
                 operations.Remove(operation);
             }
-           
+        }
+
+        private class StorageOperationComparer : IComparer<IStorageOperation>
+        {
+            private readonly Type[] _topologicallyOrderedTypes;
+
+            public StorageOperationComparer(Type[] topologicallyOrderedTypes)
+            {
+                _topologicallyOrderedTypes = topologicallyOrderedTypes;
+            }
+
+            public int Compare(IStorageOperation x, IStorageOperation y)
+            {
```

Comment:
Might be worth check Reference equals (for performance reasons).

Reply:
Done